src/components/new-safe/create/steps/ReviewStep/index.tsx

```diff
@@ -147,40 +170,82 @@ const ReviewStep = ({ data, onSubmit, onBack, setStep }: StepRenderProps<NewSafe
 
   const totalFee = getTotalFeeFormatted(maxFeePerGas, gasLimit, chain)
 
-  // Only 1 out of 1 safe setups are supported for now
-  const isCounterfactual = data.threshold === 1 && data.owners.length === 1 && isCounterfactualEnabled
+  const allSafes = useAllSafes()
+  const knownAddresses = useMemo(() => uniq(allSafes?.map((safe) => safe.address)), [allSafes])
+
+  const customRPCs = useAppSelector(selectRpc)
 
   const handleBack = () => {
     onBack(data)
   }
 
-  const createSafe = async () => {
-    if (!wallet || !chain) return
+  const handleCreateSafeClick = async () => {
+    try {
+      if (!wallet || !chain) return
 
-    setIsCreating(true)
+      setIsCreating(true)
 
-    try {
+      // Create universal deployment Data across chains:
       const readOnlyFallbackHandlerContract = await getReadOnlyFallbackHandlerContract(data.safeVersion)
+      const safeL2Deployment = getSafeL2SingletonDeployment({ version: data.safeVersion, network: chain.chainId })
+      const safeL2Address = safeL2Deployment?.defaultAddress
+      if (!safeL2Address) {
+        throw new Error('No Safe deployment found')
+      }
 
       const props: DeploySafeProps = {
         safeAccountConfig: {
           threshold: data.threshold,
           owners: data.owners.map((owner) => owner.address),
           fallbackHandler: await readOnlyFallbackHandlerContract.getAddress(),
+          to: SAFE_TO_L2_SETUP_ADDRESS,
```

Comment:
If the network / safe version is only 1.3.0 or lower we cannot use the setupToL2 flow.